app/controllers/projects_controller.rb

```diff
@@ -0,0 +1,9 @@
+class ProjectsController < ApplicationController
+
+  def show 
+    @project = Project.find(params[:id])
+    if params[:contestant_id]
```

Comment:
With this setup, your show action is doing a bit too much. Even if adding a contestant to a project will eventually redirect back to that project show page, that doesn't mean this action has to handle adding a contestant to project. 

A more RESTful option could be to have that form route as a POST request to a path like "/projects/:project_id/contestants" and be directed to the ProjectContestantsController #create action. This path indicates that you're associated a project and contestant together because both are present in the path, and then a create action handles the data operation since you're creating a joins record. THEN, you can redirect back to "/applications/:application_id"

Reply:
Thank you Juliet for all the feedback. I guess my main question that applies to all of these is if any (or all) of these are failing violations so that I can pay particular attention to them. 

I believe I understand the purpose of each piece of feedback, but I'd like to know what to prioritize, or if none of them are failing violation I can make the decision on what to prioritize.

Reply:
Good question! I'd prioritize this RESTfulness comment, the MVC infraction  with `.count`, and using Ruby in avg_experience.